### ISSUE TITLE
Can't Parameterize Molecules With Zero Partial Charges

### ISSUE THREAD

**Issue by SimonBoothroyd** [openforcefield/openforcefield#381]
**Describe the bug**
Trying to parameterize a molecule which should have zero partial charges will raise an exception.

**To Reproduce**
```
from openforcefield.topology import Molecule, Topology
from openforcefield.typing.engines.smirnoff import ForceField

molecule = Molecule.from_smiles('N#N')
topology = Topology.from_molecules([molecule])

force_field = ForceField('smirnoff99Frosst.offxml')
force_field.create_openmm_system(topology)
```

**Output**
```
    force_field.create_openmm_system(topology)
  File "/XXX/lib/python3.7/site-packages/openforcefield/typing/engines/smirnoff/forcefield.py", line 1128, in create_openmm_system
    parameter_handler.create_force(system, topology, **kwargs)
  File "/XXX/lib/python3.7/site-packages/openforcefield/typing/engines/smirnoff/parameters.py", line 2447, in create_force
    temp_mol.compute_partial_charges_am1bcc(toolkit_registry=toolkit_registry)
  File "/XXX/lib/python3.7/site-packages/openforcefield/topology/molecule.py", line 1991, in compute_partial_charges_am1bcc
    self
  File "/XXX/lib/python3.7/site-packages/openforcefield/utils/toolkits.py", line 2964, in call
    return method(*args, **kwargs)
  File "/XXX/lib/python3.7/site-packages/openforcefield/utils/toolkits.py", line 1252, in compute_partial_charges_am1bcc
    "Partial charge calculation failed. Charges from compute_partial_charges() are all 0."
Exception: Partial charge calculation failed. Charges from compute_partial_charges() are all 0.
```

**Computing environment (please complete the following information):**
 - Operating system: MacOS
 - Toolkit Version: 0.4.0

**Comment by davidlmobley** [comment]
That's slightly tricky since normally we want to make sure partial charges are not all zero. 🤔 

Maybe we want to switch to checking that the partial charge calculation ran and assigned charges, but not check whether they are nonzero...? Or, check that the partial charges are nonzero except if they should be zero (not sure how to check if they should be zero though; it's possible one could do this using symmetry groups somehow, or otherwise just have a list of cases where charges should be zero...?).

**Comment by dotsdl** [comment]
Hey @trevorgokey, I committed to taking this issue on along with #305 and #379. Is there any progress you'd like to report to help me get started, or is this perhaps complete?

**Comment by j-wags** [comment]
>>  Maybe we want to switch to checking that the partial charge calculation ran and assigned charges, but not check whether they are nonzero...?

> Is it enough (and possible) to check that partial charge calculation and assignment occurred without checking that they are all nonzero?

I think that this is now possible -- As of the addition of `LibraryChargeHandler` (before the 0.6.0 release), I added a check for unassigned and nonintegral charges, and I started recording which method generated each molecule's charges on a copy of the Topology. So, during the final `allow_nonintegral_charges` check, we should be able to allow situations where a molecule's charges are BOTH all-zero AND assigned by `charge_from_molecules` or a `LibraryCharge`.

**Comment by dotsdl** [comment]
It would appear that this may already have everything it needs final check-wise in place, by way of `ElectrostaticsHandler.postprocess_system`. This relies however on handlers like `ToolkitAM1BCCHandler` doing their job and not silently failing to assign charges to the molecules it is passed.

As for whether charges were assigned by the call to `quacpac` in `OpenEyeToolkitWrapper.compute_partial_charges_am1bcc`, is there ever a case where [`quacpac_status`](https://github.com/openforcefield/openforcefield/blob/master/openforcefield/utils/toolkits.py#L1492) will be `True` and charging by `quacpac` failed? If not, I think it's safe to remove the "all charges = 0" check at the end with no loss in safety.

**Comment by dotsdl** [comment]
I have added #574 with the proposed approach. If this sounds satisfactory, I will finish that out with changelog entry, test, etc.

**Comment by dotsdl** [comment]
Closing, as this is addressed by #574.
